awx/main/access.py

```diff
@@ -2764,5 +2766,67 @@ def can_delete(self, obj):
         return False
 
 
+class WorkflowApprovalAccess(BaseAccess):
+    '''
+    I can approve workflows when:
+     - I'm authenticated
+    I can create when:
+     - I'm a superuser:
+    '''
+
+    model = WorkflowApproval
+    prefetch_related = ('created_by', 'modified_by',)
+
+    def can_read(self, obj):
+        return True
+
+    def can_use(self, obj):
+        return True
+
+    def can_start(self, obj, validate_license=True):
+        return True
+
+    def filtered_queryset(self):
+        return self.model.objects.filter(
+            unified_job_node__in=WorkflowJobNode.accessible_pk_qs(
+                self.user, 'read_role'))
+
+    def get_queryset(self):
+        return super(WorkflowApprovalAccess, self).get_queryset().exclude(
+            workflow_approval_template__isnull=False)
+
+    def can_approve_or_deny(self, obj):
+        if self.user.approval_role or self.user.system_administrator:
```

Comment:
This probably needs to be something like:

```python
if self.user in self.approval_role or self.user.roles.filter(singleton_name=ROLE_SINGLETON_SYSTEM_ADMINISTRATOR).exists():
```

Reply:
👍 for `self.user in obj.approval_role`

for the other bit, you should use `self.user.is_superuser`, or the decorator